src/main/java/tahafurkan/sandbox/familybudgetapp/controller/SpendingController.java

```diff
@@ -15,11 +17,17 @@ public class SpendingController {
     SpendingService spendingService;
 
     @GetMapping(value = "/spendings")
-    public List<Spending> getAll(){
+    public List<Spending> getAll() {
         return spendingService.getAll();
     }
+
+    @GetMapping(value = "/spendings", params = {"spendingByMonth"})
+    public List<Spending> getSpendingsByMonth(@RequestParam String spendingByMonth) {
```

Comment:
Gelen string degil bir tarih formati olmali. Yani client tarafi sectigi bir ay icin veya 3 ay icin veya 1 sene icin sorgu yapabilir. Ama burasi icin gelen istek hep ayni sekilde olur. Bir tarih araligindaki kayitlar.
Daha once bir yere daha yazmistim, buraya bir tarih geldiginde bunun tek basina bir anlami yok. Mesela "Aralik" geldi buraya. 2021'in Aralik ayi, 2022'nin Aralik ayi, hangisi nereden bilecegiz?